nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -1577,6 +1743,12 @@ mod tests {
 
         for (table_name, result) in [
             query_count!(blueprint, id),
+            query_count!(bp_sled_state, blueprint_id),
+            query_count!(bp_sled_omicron_datasets, blueprint_id),
+            query_count!(bp_sled_omicron_physical_disks, blueprint_id),
+            query_count!(bp_sled_omicron_zones, blueprint_id),
+            query_count!(bp_omicron_dataset, blueprint_id),
+            query_count!(bp_omicron_physical_disk, blueprint_id),
```

Comment:
I'm checking that we have successfully deleted these rows here -- notably, I added checks for some parts of the blueprint that aren't related to this PR, but we should be testing them too!